infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -946,6 +1001,129 @@ private AddressMetaData writeRecord(SegmentHandle fh, long address,
         return new AddressMetaData(metadata.getChecksum(), metadata.getLength(), channelOffset);
     }
 
+    long getSegment(LogData entry) {
+        return entry.getGlobalAddress() / RECORDS_PER_LOG_FILE;
+    }
+
+    /**
+     * Pre-process a range of entries to be written. This includes
+     * removing trimmed entries and advancing the trim mark appropriately.
+     * @param range range of entries
+     * @return A subset of range; entries to be written
+     */
+    List<LogData> preprocess(List<LogData> range) {
+        List<LogData> processed  = new ArrayList<>();
+        for (int x = 0; x < range.size(); x++) {
+            // TODO(Maithem) Add an extra check to make
+            // sure that trimmed entres don't alternate
+            // with non-trimmed entries
+            LogData curr = range.get(x);
+            if (curr.isTrimmed()) {
+                // We don't need to write trimmed entries
+                // because we already track the trim mark
+                prefixTrim(curr.getGlobalAddress());
+                continue;
+            } else if (isTrimmed(curr.getGlobalAddress())) {
+                // This is the case where the client tries to
+                // write a range of addresses, but before the
+                // request starts, a prefix trim request executes
+                // on this logging unit, as a consequence trimming
+                // all of, or part of the range.
+                continue;
+            }
+            else {
+                processed.add(curr);
+            }
+        }
+        return processed;
+    }
+
+    /**
+     * This method verifies that a range of entries doesn't
+     * span more than two segments and that the log addresses
+     * are ordered sequentially.
+     * @param range entries to verify
+     * @return return true if the range is valid.
+     */
+    boolean verify(List<LogData> range) {
+
+        // Make sure that entries are ordered sequentially
+        long firstAddress = range.get(0).getGlobalAddress();
+        for (int x = 1; x < range.size(); x++) {
+            if (range.get(x).getGlobalAddress() !=
+                    firstAddress + x) {
+                return false;
+            }
+        }
+
+        // Check if the range spans more than two segments
+        long lastAddress = range.get(range.size() - 1).getGlobalAddress();
+        long firstSegment = firstAddress / RECORDS_PER_LOG_FILE;
+        long lastSegment = lastAddress / RECORDS_PER_LOG_FILE;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "lastSegment" which hides the field declared at line 92. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)